### PR TITLE
nested properties support

### PR DESCRIPTION
**Pull request by a-wart**
```json
{"city_route":
      { "airline": "Fluglinie"}
}
```

```js
console.log(__('city_route.airline')); // Fluglinie
```

### CODE REVIEW

**Comment by brandonbloom** [comment]
This assumes that keys are intended to be valid identifiers, but the existing usage supports arbitrary string keys.

**Comment by elfet** [comment]
That's ok. You can use it as you want.

**Comment by elfet** [comment]
ping  @sokra

File: index.js [hunk 1]
```diff
@@ -72,11 +72,34 @@ I18nPlugin.prototype.apply = function(compiler) {
 
 /**
  *
- * @param {object}	localization
+ * @param {object}  localization
+ * @param {string}  string key
+ * @returns {*}
+ */
+function byString(object, stringKey) {
+    stringKey = stringKey.replace(/\[(\w+)\]/g, '.$1'); // convert indexes to properties
```
Comment (sokra) [inline index.js]: Either don't allow this at all or implement it correctly.

`abc[def]` is not the correct syntax. `abc["def"]` is the correct syntax. Also make sure that `abc["def['ghi']"]` works ;).

File: index.js [hunk 2]
```diff
@@ -72,11 +72,34 @@ I18nPlugin.prototype.apply = function(compiler) {
...
+}
+
+/**
+ *
+ * @param {object}  localization
  * @returns {Function}
  */
 function makeLocalizFunction(localization) {
-	return function localizFunction(key) {
-		return localization[key];
```
Comment (sokra) [inline index.js]: spacing... Only use tabs. Feel free to copy the `.editorconfig` file from the webpack repo.

File: index.js [hunk 3]
```diff
@@ -72,11 +72,34 @@ I18nPlugin.prototype.apply = function(compiler) {
...
+
+    return object;
+}
+
+/**
+ *
+ * @param {object}  localization
  * @returns {Function}
  */
 function makeLocalizFunction(localization) {
```
Comment (sokra) [inline index.js]: You may fix the typo in the function name too.

**Comment by sokra** [comment]
This would be a breaking change. It's propably better to make this opt-in.

Try the change the signature of the plugin to `Plugin(localization, options)` (plus backward-compatiblility) and add an `nested` property.

**Comment by vinkla** [comment]
:+1: on adding support for nested values.

**Comment by ronkorving** [comment]
:+1: from me too. I really wanna use this plugin, but it seems rather... dead and missing some key features. Should I roll my own, or is this coming back to life any time soon? :)

**Comment by elfet** [comment]
@wart what about PR?

**Comment by a-wart** [comment]
coming soon guys, just couple of days:)

**Comment by ronkorving** [comment]
I feel like a kid in a candy store 😊

**Comment by a-wart** [comment]
@sokra I updated PR, check it please

File: index.js [hunk 4]
```diff
@@ -72,11 +84,33 @@ I18nPlugin.prototype.apply = function(compiler) {
...
+		stringKey = stringKey.replace(/^\./, ''); // strip a leading dot
+
+		var keysArray = stringKey.split('.');
+		for (var i = 0, length = keysArray.length; i < length; ++i) {
+			var key = keysArray[i];
+				if (key in object) {
+					object = object[key];
+				} else {
+					return;
+				}
```
Comment (sokra) [inline index.js]: Too much spacing

File: index.js [hunk 5]
```diff
@@ -8,17 +8,29 @@ var MissingLocalizationError = require("./MissingLocalizationError");
...
+			functionName: options
+		};
+	}
+
+	if (arguments[2]) {
+		options.failOnMissing = arguments[2];
+	}
+
+	this.options = options || {};
+	this.localization = localization? ('function' === typeof localization? localization: makeLocalizeFunction(localization, !!options.nested))
```
Comment (sokra) [inline index.js]: please make it consitent and use `this.options` instead of `options`

**Comment by a-wart** [comment]
updated

**Comment by scythargon** [comment]
@sokra

**Comment by sokra** [comment]
Thanks

**Comment by ronkorving** [comment]
@sokra and thank you!

**Pull request merged**